components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMService.java

```diff
@@ -115,10 +117,13 @@ public Application getApplicationByName(String clientName) throws DCRMException
             }
             return buildResponse(oAuthConsumerAppDTO);
         } catch (IdentityOAuthAdminException e) {
+            if (INVALID_OAUTH_CLIENT.getErrorCode().equals(e.getErrorCode())) {
+                throw DCRMUtils.generateClientException(
+                        DCRMConstants.ErrorMessages.NOT_FOUND_APPLICATION_WITH_NAME, clientName);
```

Comment:
since this error code is to check if any application exists, let's introduce a different error code to handle the scenario where the application is not an oauth app

Reply:
fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2142/commits/291575be36abf83773391824bdfbe980de4fc8ef